### PR TITLE
Add test for ignoring blank/whitespace VM options

### PR DESCRIPTION
**Pull request by sharon-wang**
For PR #5353 (Ignore empty and whitespace VM options)

This test checks that creating VMs from C with blank or whitespace options produces the appropriate results for the following cases:
- Ignore whitespace VM option (disregard option and start VM)
- Ignore empty VM option (disregard option and start VM)
- Reject whitespace VM option (indicate unrecognized option and do not start VM)
- Reject empty VM option (indicate unrecognized option and do not start VM)

Extensions changes: 
- JDK: https://github.com/ibmruntimes/openj9-openjdk-jdk/pull/117
- JDK 8: https://github.com/ibmruntimes/openj9-openjdk-jdk8/pull/309
- JDK 11: https://github.com/ibmruntimes/openj9-openjdk-jdk11/pull/184
- JDK 12: https://github.com/ibmruntimes/openj9-openjdk-jdk12/pull/57

Signed-off-by: Sharon Wang <sharon-wang-cpsc@outlook.com>

### CODE REVIEW

**Comment by sharon-wang** [comment]
> Looks good to me. I suggest adding a test that has multiple extra white space characters.

Added a couple test cases for multiple whitespace characters

File: runtime/tests/redirector/nativevmargs/testnativevmargs.c [hunk 1]
```diff
@@ -0,0 +1,182 @@
...
+		fprintf(stderr, "[ERR] Failed memory allocation for JVM arguments.\n");
+		rc = 2;
+		goto fail;
+	}
+
+	/* Create the JVM options */
+	for (int i = 2; jvmOptionCount < optionCount; i++) {
+		if (0 == strncmp(argv[i], "-Custom:", 8)) {
+			/* convert custom options to JVM options */
+			if (0 == strcmp(argv[i], "-Custom:WhitespaceOption")) {
```
Comment (pshipton) [inline runtime/tests/redirector/nativevmargs/testnativevmargs.c]: Do we still need to test with a single space now that MultipleWhitespaceOption is added? Running command line tests are a little costly, more so on certain platforms, since the JVM needs to start up each time.

File: test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    <variable name="TEST_NATIVEVMARGS_CONCLUDED_WITH_CODE" value="[MSG] Test nativevmargs concluded with code:"/>
+
+    <test id="test nativevmargs ignore whitespace VM option">
+        <command>$NATIVEVMARGS$ $JVMLIBPATH$ $IGNORE_UNRECOGNIZED_OPTIONS$ $WHITESPACE_OPTION$</command>
+        <output type="success" caseSensitive="yes" regex="no">$TEST_NATIVEVMARGS_CONCLUDED_WITH_CODE$ 0</output>
+        <output type="failure" regex="no">[ERR]</output>
+    </test>
+
+    <test id="test nativevmargs ignore empty VM option">
+        <command>$NATIVEVMARGS$ $JVMLIBPATH$ $IGNORE_UNRECOGNIZED_OPTIONS$ $EMPTY_OPTION$</command>
```
Comment (pshipton) [inline test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: Can we combine the EMPTY_OPTION test with MULTIPLE_WHITESPACE_OPTION, for the passing case, to reduce the testing cost?

Comment (sharon-wang) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: For sure! I will remove the EMPTY_OPTION test since the MULTIPLE_WHITESPACE_OPTION test should cover the passing case.

Comment (pshipton) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: Not exactly what I meant. The passing case can test with both options `$EMPTY_OPTION$` and `$MULTIPLE_WHITESPACE_OPTION$` on the command line.

File: runtime/tests/redirector/nativevmargs/testnativevmargs.c (hunk 1 above)
Comment (sharon-wang) [reply runtime/tests/redirector/nativevmargs/testnativevmargs.c]: MultipleWhitespaceOption will cover the single space case - will be removing single space tests

File: test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml (hunk 2 above)
Comment (sharon-wang) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: Ah I see, I'll make that change

Comment (sharon-wang) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: I think the tests may need to be separate, since it looks like `checkArgsConsumed()` will return as soon as the first unrecognized option is encountered: https://github.com/eclipse/openj9/blob/567bdff454ec19d0bf78c4272921b27d9d5ffdab/runtime/vm/jvminit.c#L2386-L2449

Comment (pshipton) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: That's why we can't combine the case where `COMMANDLINE_OPTION_UNRECOGNIZED` is expected, but we can combine the (passing) case where the arguments are expected to be ignored as if either the empty option or the whitespace option isn't ignored this test will fail. I didn't explain what I meant by the "passing case" very well, by which I meant the test with `IGNORE_UNRECOGNIZED_OPTIONS` when `$TEST_NATIVEVMARGS_CONCLUDED_WITH_CODE$ 0` is expected.

Comment (sharon-wang) [reply test/functional/cmdLineTests/cmdLineTest_J9tests/j9tests_native.xml]: I think I'm on the same page now 🙂 There is now one combined passing test case for empty and whitespace, and two separate tests for empty and whitespace for when `COMMANDLINE_OPTION_UNRECOGNIZED` is expected.

**Comment by pshipton** [comment]
jenkins test extended plinux jdk8 depends ibmruntimes/openj9-openjdk-jdk8#309

**Comment by pshipton** [comment]
@theresa-m are you still good with the latest changes?

**Comment by theresa-m** [comment]
works for me :)

**Comment by pshipton** [comment]
This is failing to compile on Windows, reverting to sort that out.
```
15:09:31  testnativevmargs.c
15:09:31  testnativevmargs.c(97) : error C2143: syntax error : missing ';' before 'type'
15:09:31  testnativevmargs.c(97) : error C2143: syntax error : missing ';' before 'type'
15:09:31  testnativevmargs.c(97) : error C2143: syntax error : missing ')' before 'type'
15:09:31  testnativevmargs.c(97) : error C2143: syntax error : missing ';' before 'type'
15:09:31  testnativevmargs.c(97) : warning C4552: '<' : operator has no effect; expected operator with side-effect
15:09:31  testnativevmargs.c(97) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(97) : error C2059: syntax error : ')'
15:09:31  testnativevmargs.c(97) : error C2143: syntax error : missing ';' before '{'
15:09:31  testnativevmargs.c(98) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(100) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(102) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(104) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(107) : error C2044: illegal continue
15:09:31  testnativevmargs.c(109) : error C2065: 'i' : undeclared identifier
15:09:31  testnativevmargs.c(114) : error C2065: 'i' : undeclared identifier
```

**Comment by pshipton** [comment]
@sharon-wang please fix the Windows problem and create a PR with the changes. I'll be sure to include Windows PR testing before merging again.
